### ISSUE TITLE
Remove os.stat_float_times()

### ISSUE THREAD

**Issue by vstinner** [python/cpython#76008]
BPO | [31827](https://bugs.python.org/issue31827)
--- | :---
Nosy | @vstinner, @serhiy-storchaka
PRs | <li>python/cpython#4061</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-10-24.09:19:23.678>
created_at = <Date 2017-10-20.14:03:40.126>
labels = ['3.7', 'library']
title = 'Remove os.stat_float_times()'
updated_at = <Date 2017-10-24.09:19:23.677>
user = 'https://github.com/vstinner'
```

bugs.python.org fields:
```python
activity = <Date 2017-10-24.09:19:23.677>
actor = 'vstinner'
assignee = 'none'
closed = True
closed_date = <Date 2017-10-24.09:19:23.678>
closer = 'vstinner'
components = ['Library (Lib)']
creation = <Date 2017-10-20.14:03:40.126>
creator = 'vstinner'
dependencies = []
files = []
hgrepos = []
issue_num = 31827
keywords = ['patch']
message_count = 6.0
messages = ['304655', '304660', '304719', '304875', '304881', '304882']
nosy_count = 2.0
nosy_names = ['vstinner', 'serhiy.storchaka']
pr_nums = ['4061']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue31827'
versions = ['Python 3.7']
```

</p></details>

**Comment by vstinner** [comment]
os.stat_float_times() was introduced in Python 2.3 to get file modification times with sub-second resolution. The default remains to get time as seconds (integer). See commit f607bdaa77475ec8c94614414dc2cecf8fd1ca0a.

The function was introduced to get a smooth transition to time as floating point number, to keep the backward compatibility with Python 2.2.

In Python 2.5, os.stat() returns time as float by default: commit fe33d0ba87f5468b50f939724b303969711f3be5.

Python 2.5 was released 11 years ago. I consider that people had enough time to migrate their code to float time :-)

I modified os.stat_float_times() to emit a DeprecationWarning in Python 3.1: commit 034d0aa2171688c40cee1a723ddcdb85bbce31e8 (bpo-14711).

**Comment by vstinner** [comment]
Attached [PR 4061](https://github.com/python/cpython/pull/4061) removes os.stat_float_times().

**Comment by serhiy-storchaka** [comment]
stat_result is a named 10-tuple, containing several additional attributes. The last three items are st_atime, st_mtime and st_ctime as integers. Accessing them by name returns floats. Isn't a time to make them floats when access stat_result as a tuple?

**Comment by vstinner** [comment]
New changeset 01b5aab7bfb11ee5476ef52d24495598cbe7c99a by Victor Stinner in branch 'master':
bpo-31827: Remove os.stat_float_times() (GH-4061)
https://github.com/python/cpython/commit/01b5aab7bfb11ee5476ef52d24495598cbe7c99a

**Comment by vstinner** [comment]
Copy of my comment on the PR: "So, I tried to remove the backward compatibility layer: I modified stat_result[ST_MTIME] to return float rather than int. Problem: it broke test_logging, the code deciding if a log file should be rotated or not.

While I'm not strongly opposed to modify stat_result[ST_MTIME], I prefer to do it in a separated PR. Moreover, we need maybe to emit a DeprecationWarning, or at least deprecate the feature in the doc, before changing the type, no?"

Serhiy's answer: "I agree, it should be done in a separate issue. It needs a special discussion. And maybe this can't be changed."
